### PR TITLE
platform(general): use sca inline suppressions

### PR DESCRIPTION
**Pull request by AdamDev**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Use SCA Inline suppression that was extracted in the cloud via the static analysis process.
## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/sca/output.py [hunk 1]
```diff
@@ -266,6 +276,35 @@ def _add_to_report_licenses_statuses(
...
+        return None
+    suppressions_map: _ScaSuppressionsMaps = {}
+
+    # fill cves suppressions map
+    cve_by_cve_map: dict[str, _SuppressedCves] = {}
+    if inline_suppressions.get("cves"):
+        if inline_suppressions["cves"].get("byCve"):
+            for cve_suppression in inline_suppressions["cves"]["byCve"]:
+                if cve_suppression.get("cveId"):
+                    cve_by_cve_map[cve_suppression["cveId"]] = cve_suppression
```
Comment (gruebel) [inline checkov/common/sca/output.py:290]:
```suggestion
    inline_suppressions_by_cve = inline_suppressions.get("cves", {}).get("byCve", {})
    for cve_suppression in inline_suppressions_by_cve:
        cve_id = cve_suppression.get("cveId")
        if cve_id:
            cve_by_cve_map[cve_id] = cve_suppression
```
🙂

File: checkov/common/sca/output.py [hunk 2]
```diff
@@ -266,6 +276,35 @@ def _add_to_report_licenses_statuses(
     return licenses_per_package_map
 
 
+def get_inline_suppressions_map(inline_suppressions: _ScaSuppressions | None = None) -> _ScaSuppressionsMaps | None:
+    if not inline_suppressions:
+        return None
+    suppressions_map: _ScaSuppressionsMaps = {}
+
+    # fill cves suppressions map
+    cve_by_cve_map: dict[str, _SuppressedCves] = {}
```
Comment (gruebel) [inline checkov/common/sca/output.py:285]: then name seems a bit strange, maybe `cve_suppresion_by_cve_map`?

File: checkov/common/sca/output.py [hunk 3]
```diff
@@ -266,6 +276,35 @@ def _add_to_report_licenses_statuses(
...
+
+    # fill licenses suppressions map
+    licenses_by_policy_and_package_map: dict[str, _SuppressedLicenses] = {}
+    if inline_suppressions.get("licenses"):
+        if inline_suppressions["licenses"].get("byPackage"):
+            for license_suppression in inline_suppressions["licenses"]["byPackage"]:
+                if license_suppression.get("licensePolicy") and license_suppression.get("packageName"):
+                    key = get_license_policy_and_package_alias(license_suppression["licensePolicy"],
+                                                               license_suppression["packageName"])
+                    licenses_by_policy_and_package_map[key] = license_suppression
```
Comment (gruebel) [inline checkov/common/sca/output.py:300]: you can apply similar improvements here as mentioned before

File: checkov/common/sca/output.py [hunk 4]
```diff
@@ -422,37 +462,40 @@ def add_cve_record_to_report(vulnerability_details: dict[str, Any], package_name
...
-                        record.check_result = {
-                            "result": CheckResult.SKIPPED,
-                            "suppress_comment": cve_suppression["suppress_comment"],  # type:ignore[index]
-                        }
-                        return True
-        if "cve" in inline_suppressions:
-            cve_suppression = inline_suppressions["cve"].get(
-                vulnerability_details.get("cveId", vulnerability_details.get("id", "")))
+    if inline_suppressions_maps:
+        if inline_suppressions_maps.get("cve_by_cve_map") and record.vulnerability_details:
```
Comment (gruebel) [inline checkov/common/sca/output.py:471]:
```suggestion
        if record.vulnerability_details and inline_suppressions_maps.get("cve_by_cve_map"):
```
it is faster to check if a value is set compared to a dict lookup

**Review by gruebel — COMMENTED** [review]
nice work 💪

File: checkov/common/sca/output.py [hunk 5]
```diff
@@ -422,37 +462,40 @@ def add_cve_record_to_report(vulnerability_details: dict[str, Any], package_name
...
+        if inline_suppressions_maps.get("licenses_by_policy_and_package_map"):
+            key = get_license_policy_and_package_alias(vulnerability_details.get("policy", ""),
+                                                       vulnerability_details.get("package_name", ""))
+            license_suppression = inline_suppressions_maps["licenses_by_policy_and_package_map"].get(key)
+            if license_suppression:
+                record.check_result = {
+                    "result": CheckResult.SKIPPED,
+                    "suppress_comment": license_suppression.get('reason', ''),
                 }
                 return True
```
Comment (ayajbara) [inline checkov/common/sca/output.py:500]:
```suggestion
    if inline_suppressions_maps and inline_suppressions_maps.get("licenses_by_policy_and_package_map"):
            key = get_license_policy_and_package_alias(vulnerability_details.get("policy", ""),
                                                       vulnerability_details.get("package_name", ""))
            license_suppression = inline_suppressions_maps["licenses_by_policy_and_package_map"].get(key)
            if license_suppression:
                record.check_result = {
                    "result": CheckResult.SKIPPED,
                    "suppress_comment": license_suppression.get('reason', ''),
                }
                return True
```

File: checkov/sca_package_2/runner.py [hunk 6]
```diff
@@ -122,10 +121,7 @@ def run(
                                 for elm in result.get("license_statuses") or []]
 
             rootless_file_path = str(package_file_path).replace(package_file_path.anchor, "", 1)
-            inline_suppressions = extract_inline_suppressions(
-                uploaded_files=uploaded_files,
-                rootless_file_path=rootless_file_path,
-            )
+            inline_suppressions = result.get("inlineSuppressions")
```
Comment (ayajbara) [inline checkov/sca_package_2/runner.py:124]: 🥇

**Review by ayajbara — APPROVED** [review]
💯

File: checkov/common/sca/output.py (hunk 5 above)
Comment (AdamDev) [reply checkov/common/sca/output.py]: fixed

**Review by gruebel — APPROVED** [review]
🍻

**Pull request merged**